test/powershell/Modules/Microsoft.PowerShell.Utility/Sort-Object.Tests.ps1

```diff
@@ -222,3 +222,229 @@ Describe "Sort-Object DRT Unit Tests" -Tags "CI" {
 		$results[2] | Should Be 1
 	}
 }
+
+Describe 'Sort-Object Top and Bottom Unit Tests' -Tags 'CI' {
+
+	# Helper function to compare two sort entries
+	function Compare-SortEntry
+	{
+		param($nSortEntry, $fullSortEntry)
+		if ($nSortEntry -is [System.Array]) {
+			# Arrays must be compared differently than everything else
+			$nSortEntry.Equals($fullSortEntry) | Should Be $true
+		} else {
+			$nSortEntry | Should Be $fullSortEntry
+		}
+	}
+
+	# Helper function that compares a full sort with an n-sort
+	function Test-SortObject {
+		param([array]$unsortedData, [hashtable]$baseSortParameters, [string]$nSortType, [int]$nValue)
+		$nSortParameters = @{
+			$nSortType = $nValue
+		}
+		# Sort the data
+		$fullSortResults = $unsortedData | Sort-Object @baseSortParameters
+		$nSortResults = $unsortedData | Sort-Object @baseSortParameters @nSortParameters
+		# Verify the counts when not doing a -Unique sort
+		if (-not $baseSortParameters.ContainsKey('Unique')) {
+			$nSortResults.Count | Should Be $(if ($nSortParameters[$nSortType] -gt $unsortedData.Length) {$unsortedData.Length} else {$nSortParameters[$nSortType]})
+			$fullSortResults.Count | Should Be $unsortedData.Length
+		}
+		# Compare the n-sort result entries with their corresponding full sort result entries
+		if ($nSortType -eq 'Top') {
+			$range = 0..$($nSortResults.Count - 1)
+		} else {
+			$range = -$nSortResults.Count..-1
+		}
+		foreach ($i in $range) {
+			Compare-SortEntry $nSortResults[$i] $fullSortResults[$i]
+		}
+	}
+
+	# Test cases when only the n-sort type needs to be changed
+	$topBottom = @(
+		@{nSortType='Top'   }
+		@{nSortType='Bottom'}
+	)
```

Comment:
Previously I had in mind, that these variables ($topBottom and $topBottomAscendingDescending) are good candidates to move in BeforeAll.